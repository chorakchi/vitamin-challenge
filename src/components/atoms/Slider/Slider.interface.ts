import { InputHTMLAttributes } from 'react';


export interface SliderProps
  extends InputHTMLAttributes<HTMLInputElement> {
    label?: string;
    error?: boolean;
}