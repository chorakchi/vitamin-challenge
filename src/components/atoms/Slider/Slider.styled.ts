import styled from "styled-components";

export const SliderStyled = styled.input`
-webkit-appearance: none;
  width: 300px;
  height: 8px;
  background: #d3d3d3;
  outline: none;
  opacity: 0.7;
  -webkit-transition: .2s;
  transition: opacity .2s;
  border-radius: 20px;
  margin: 20px 0;
  :hover {
  opacity: 1;
}

::-webkit-slider-thumb {
  -webkit-appearance: none;
  appearance: none;
  width: 25px;
  height: 25px;
  background: #04AA6D;
  cursor: pointer;
  border-radius: 4px;
}

::-moz-range-thumb {
  width: 25px;
  height: 25px;
  background: #04AA6D;
  cursor: pointer;
  border-radius: 4px;

}
`;
