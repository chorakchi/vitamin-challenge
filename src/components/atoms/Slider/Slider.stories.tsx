import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Slider } from "./Slider";

export default {
  title: "Atoms/Slider",
  component: Slider,
  parameters: {
    docs: {
      description: {
        component:
          `Slider let users enter and edit range number.They typically appear in forms and dialogs.
                <br/>` +
          `The Slider  component is a complete form control including a label, input. <br/>` +
          `<br/> <br/> <h5> Form props:</h5>` +
          " Standard form attributes are supported e.g. ```required```, ```disabled```, ```type```, etc. <br/> <br/>"
      },
    },
    backgrounds: {
      default: "light",
    },
  },
} as ComponentMeta<typeof Slider>;

const Template: ComponentStory<typeof Slider> = (args) => (
  <Slider {...args} />
);

export const Slider_ = Template.bind({});
Slider_.args = {};
