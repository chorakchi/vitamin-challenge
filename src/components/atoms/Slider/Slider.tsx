import { SliderStyled } from "./Slider.styled";
import { SliderProps } from "./Slider.interface";
import { Typography } from "../Typography/Typography";
export const Slider: React.FC<SliderProps> = ({
  label = "Url:",
  error,
  ...props
}: SliderProps) => {
  return (
    <span>
      <Typography as="label" htmlFor={props.id}>
        {label}
        <br/>
        <SliderStyled type="range" {...props} />
      </Typography>
    </span>
  );
};
