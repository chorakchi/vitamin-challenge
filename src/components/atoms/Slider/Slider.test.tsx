import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import {Slider} from "./Slider";

describe("Slider", () => {
  it("should render the Slider", () => {
    const subject = shallow(<Slider />);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer.create(<Slider />).toJSON();
    expect(subject).toMatchSnapshot();
  });
});