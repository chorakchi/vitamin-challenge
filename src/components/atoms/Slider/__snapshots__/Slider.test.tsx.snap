// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should render correctly and match to the Snapshot 1`] = `
<span>
  <label
    className="Typographystyled__TypographyStyled-sc-i22hyo-0 iKdygM"
  >
    Url:
    <br />
    <input
      className="Sliderstyled__SliderStyled-sc-fj8kj3-0 ktJaBw"
      type="range"
    />
  </label>
</span>
`;
